mula/scheduler/connectors/listeners/listeners.py

```diff
@@ -105,10 +144,7 @@ def callback(
         queue.
         """
         self.logger.debug("Received message on queue %s, message: %r", method.routing_key, body)
-
-        self.dispatch(body)
-
-        channel.basic_ack(method.delivery_tag)
+        self.executor.submit(self.dispatch, channel, method.delivery_tag, body)
```

Comment:
With this setup, every class extending this parent is responsible for not forgetting to ack its messages after execution. Given the examples for `raw_data.py` and `scan_profile.py`, perhaps we should add this here in the parent class? Then you also don't have to change the signature of the `dispatch` method, e.g.:

```suggestion
        def _dispatch_and_ack(channel, delivery_tag, body: bytes):
            self.dispatch(body)
            self.connection.add_callback_threadsafe(functools.partial(self.ack_message, channel, delivery_tag))

        self.executor.submit(_dispatch_and_ack, channel, method.delivery_tag, body)
```